tmk_core/common/keyboard.c

```diff
@@ -64,20 +64,41 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 
 #ifdef MATRIX_HAS_GHOST
+static uint16_t matrix_ghost_check[MATRIX_ROWS];
```

Comment:
16 bytes is still quite much when you only have 2.5 KB available.  The easiest way would probably be to build the array on the stack each time the function is called. Or better build each row when needed, so two loops inside each other. 

I guess the program memory usage would be around the same, as the `make_ghost_check_array` function is no longer needed.